### PR TITLE
Add attachment option type for slash commands

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

This PR implements option type `11` (attachments), which can be used by annotating a parameter with `disnake.Attachment`.

Interaction handling has been tested using the payload structure shown in [this video](https://cdn.discordapp.com/attachments/889939268733132870/921494760227741706/attachments_desktop.mp4) ([source](https://discord.com/channels/613425648685547541/889939268733132870/921494800996372540)).

## Checklist

- [x] If code changes were made then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `black .`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/ext/commands/params.py [hunk 1]
```diff
@@ -320,7 +323,7 @@ async def get_default(self, inter: CommandInteraction) -> Any:
     async def verify_type(self, inter: CommandInteraction, argument: Any) -> Any:
         """Check if a type of an argument is correct and possibly fix it"""
         # these types never need to be verified
-        if self.discord_type.value in [3, 4, 5, 8, 9, 10]:
+        if self.discord_type.value in [3, 4, 5, 8, 9, 10, 11]:
```
Comment (EQUENOS) [inline disnake/ext/commands/params.py:326]: Maybe we should use a set here

**Review by EQUENOS — COMMENTED** [review]
Neat

File: disnake/ext/commands/params.py (hunk 1 above)
Comment (thesadru) [reply disnake/ext/commands/params.py:326]: we should just get rid of this entire method tbh

File: disnake/interactions/application_command.py [hunk 2]
```diff
@@ -371,7 +371,7 @@ class ApplicationCommandInteractionDataResolved:
         IDs and messages
     """
 
-    __slots__ = ("members", "users", "roles", "channels", "messages")
+    __slots__ = ("members", "users", "roles", "channels", "messages", "attachments")
```
Comment (Victorsitou) [inline disnake/interactions/application_command.py:374]: The documentation for `attachment` is missing above.

**Review by Victorsitou — COMMENTED** [review]
Also, the documentation of `OptionType.attachment` is also missing (I know it was added a some weeks ago)

File: disnake/interactions/application_command.py (hunk 2 above)
Comment (shiftinv) [reply disnake/interactions/application_command.py:370]: Fixed in f77f5a7de6d24433eac90792bb634367367e12c3, thanks

**Review by onerandomusername — APPROVED** [review]
looks good to me. Tested and read through, no problems that I can see. Still a few bugs, but they're on discord's end and shouldn't affect us.

**Review by m1raynee — COMMENTED** [review]
LGTM. Tested it and it's working. Great job.

**Review by m1raynee — APPROVED** [review]

**Review by EQUENOS — APPROVED** [review]

**Pull request merged**